### ISSUE TITLE
Enchanter cycling through levels in input and not in output.

### ISSUE THREAD

**Issue by LumberWizard** [SleepyTrousers/EnderIO#5133]
#### Issue Description:
Follow up on conversation on Discord. The issue is actually a lot simple than I remembered - when looking up an enchanted book recipe, the output book will stay the same level you pressed R on, while the inputs will cycle from level 1 to the maximum.
![EnderIO Bug](https://user-images.githubusercontent.com/5213262/58006869-b0987800-7af1-11e9-8d9a-5af7369fbdac.PNG)
While searching for uses for items used to enchant, or uses of the enchanter, cycling behaves properly.

#### What happens:
JEI cycles through the amounts of input items that correspond to different enchantment level in the output book, while the output book and its level stay the same.

#### What you expected to happen:
I expected that JEI wouldn't cycle through the amounts of the inputs while keeping the output the same.

#### Steps to reproduce:
1. Install JEI, EnderCore, and Ender IO
2. Press R on said enchanted book with an enchanter recipe.

____
#### Affected Versions (Do *not* use "latest"):

- EnderIO: 5.0.44
- EnderCore: 0.5.57
- JEI: 4.15.0.279
- Minecraft: 1.12
- Forge: 14.23.5.2838
- SpongeForge? No
- Optifine? No
- Single Player and/or Server? Both

**Comment by tyler489** [comment]
For clarification the text changes.. the actuall item doesn't?

**Comment by MatthiasMann** [comment]
From what I saw the inputs (the slot which decides the enchantment) and the output slot cycle independent from each other when looking up all recipes.
And the output stays fixed when looking up a specific enchant but the input still cycles.

I forgot if the XP and lapis costs change too or not.

**Comment by HenryLoenwind** [comment]
Yes, JEI does not have support for cycling recipes, only for cycling single slots. The recipes we provide have slots that cycle in sync because they have the same number of items in them, but as soon as JEI modifies one of those lists, that breaks.

The issue here is that we need to implement JEI's focus handling ourselves for each and every recipe, and that it cannot be done in the recipe itself (there it'd be easy) but on the already processed data that goes towards the rendering. The major recipes have this in place, but as long as it keeps breaking there, the "smaller" recipes don't get done...

**Comment by LumberWizard** [comment]
Matthias, he XP and lapis costs do change, together with the material costs. I haven't found that when looking up all recipes the outputs and inputs change independently from one another in my testing, but I believe during play a few weeks ago I have encountered that.

Tyler, the XP cost, material cost and lapis cost all cycle together with one another, and normally it's also together with the output item, except when looking up a recipe for a specific book. However, as Matthias said there could be cases where both cycle independently of one another that I haven't found in my testing yesterday.

**Comment by HenryLoenwind** [comment]
Why do people like to always chat along in those tickets where we know exactly what's happening? ;) ;)

**Comment by mindforger** [comment]
because people are curious and have ideas or exerience with something in particular on their own 

and sometimes it even helps :P

**Comment by HenryLoenwind** [comment]
Note that this is not a final fix, just a workaround to prevent invalid recipes to be shown. For now, we'll just ignore the focus and show all variants, keeping cycling through the different levels of the shown enchantment.
